hazelcast/src/main/java/com/hazelcast/instance/GroupProperties.java

```diff
@@ -300,6 +300,11 @@ public GroupProperties(Config config) {
                 properties[groupProperty.ordinal()] = propertyValue;
                 continue;
             }
+            GroupProperty parent = groupProperty.getParent();
+            if (parent != null) {
+                properties[groupProperty.ordinal()] = properties[parent.ordinal()];
```

Comment:
You cannot use an `enum` constant as parent before it's declared. So in `GroupProperty` a parent entry is always declared before its children. Furthermore the iteration is in the order of the declaration (at least the JavaDoc of enum.values() states that):
> Returns an array containing the constants of this enum type, in the order they're declared.